CMakeLists.txt

```diff
@@ -75,7 +75,23 @@ if (${CMAKE_VERSION} VERSION_GREATER "3.0.0" AND CMAKE_MAKE_PROGRAM MATCHES "nin
     set(CMAKE_JOB_POOL_LINK link_job_pool)
   endif ()
 endif ()
-  
+
+option (USE_UNITY "Use unity builds" ON) # TODO: switch to OFF before review
```

Comment:
I wanted to test if the produced binaries deliver good results on your CI.

Reply:
All the targets, which build core-tests, have succeeded in building the target via unity build. I'll leave this until you review and decide if we should maybe keep in ON by default.